cores/esp8266/uart.cpp

```diff
@@ -506,8 +506,10 @@ uart_write(uart_t* uart, const char* buf, size_t size)
 
     size_t ret = size;
     const int uart_nr = uart->uart_nr;
-    while (size--)
+    while (size--) {
         uart_do_write_char(uart_nr, pgm_read_byte(buf++));
+        optimistic_yield(10000UL);
```

Comment:
AIUI, `optimistic_yield` will `panic()` if not in the CONT context.  This would make any SYS-callback printouts (TCP, others) potentially crash.

@devyte, @d-a-v, any comments/ideas?  I like the idea, but the forced panic seems worse than a potential WDT...

Reply:
optimistic_yield() checks if it can yield (i. e. in CONT), and if enough time has elapsed since the start of the current loop. If true, it yield()s, otherwise does nothing. 

So it shouldn't panic. 

However: I seem to remember there was a change to its behavior so that yield actually only happens once every blah ms instead of after blah ms since loop start.

In any case:
- At a glance, I think the argument of 10000 is way too big, I would use something like 500, i. e. yield at most every 0.5s
- this yield won't help in the case of slow comms attempted from SYS, but you could argue that comms should be done from CONT

Reply:
The unit is `µs` so 10k is 10ms.
It should be fine.
Maybe 1ms is better to sustain high throughput in case of a network transfer.